CMakefile should have a default build type

Ensure there is a build type if user does not explicity specify one

Ok, I can reproduce:

```
cmake ../nuclei
-- Found conan executable at: /usr/local/bin/conan
'/usr/local/bin/conan' 'install' '../nuclei' '--output-folder=/tmp' '--build=missing' '--settings=build_type='
ERROR: Invalid setting '' is not a valid 'settings.build_type' value.
Possible values are ['Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel']
Read "http://docs.conan.io/2/knowledge/faq.html#error-invalid-setting"
CMake Error at cmake/ConanConfig.cmake:19 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  CMakeLists.txt:9 (include)

```

@php1ic could you look at it? maybe related: https://github.com/conan-io/conan/issues/7884

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,11 @@ set(EXTRA_MODULES_DIR ${CMAKE_CURRENT_LIST_DIR}/cmake)
 list(APPEND CMAKE_MODULE_PATH ${EXTRA_MODULES_DIR})
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
 
+#=============================================================================
+# ensure build type before calling ConanConfig
+#=============================================================================
+include(EnsureBuildType)
+
 # This MUST run before the call to project()
 # Otherwise the appropriate variables will not be set
 include(ConanConfig)
@@ -13,12 +18,6 @@ project(nuclei
   LANGUAGES CXX C
   VERSION 0.0.1
   )
-
-#=============================================================================
-# ensure build type first
-#=============================================================================
-
-include(EnsureBuildType)
 
 include(CompilerConfig)
 include(OutputDirConfig)

